release.sh

```diff
@@ -23,35 +31,57 @@ if [[ ! $VERSION =~ ^[0-9]*\.[0-9]*\.[0-9]*(-[A-Z0-9]*)?$ ]]; then
 fi
 
 GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-if [[ ! $GIT_BRANCH =~ main ]]; then
+if [[ ! $GIT_BRANCH =~ $UPSTREAM_BRANCH ]]; then
     echo "You do not seem to be on the main branch!"
     exit 1
 fi
 
-echo Releasing version "$VERSION"
+if [ -n "$(git status --porcelain)" ]; then
+    echo "You have local changes!"
+    exit 1
+fi
+
+git pull "${UPSTREAM_URL}" "${UPSTREAM_BRANCH}"
+
+RELEASE_BRANCH="${VERSION_PREFIXED}-release"
+git branch "$RELEASE_BRANCH"
+git checkout "$RELEASE_BRANCH"
 
-echo Updating version in package.json ...
+if [[ $DRY_RUN ]]; then
+  echo "Releasing version $VERSION (dry-run)"
+else
+  echo "Releasing version $VERSION"
+fi
+
+echo "Updating version in package.json ..."
 sed -i -e "s/\"version\":.*/\"version\":\ \"$VERSION\"\,/" package.json
 npm install
 
-if [ -n "$(git status --porcelain)" ]; then
-    echo Commiting version change
-    git add package.json
-    git add package-lock.json
-    git commit --signoff -m "Update version to $VERSION"
-fi
+echo "Commiting version change"
+git add package.json
+git add package-lock.json
+git commit --signoff -m "Update version to $VERSION"
 
-echo Linting, building and testing:
+echo "Linting, building and testing:"
```

Comment:
our echos have different styles, some ending in ":", some in "...", some in nothing, can we unify that?